kingpin/actors/rightscale/server_array.py

```diff
@@ -505,12 +568,17 @@ def _launch_instances(self, array, count=False):
             # Launch *up to* min_count. Not *new* min_count.
             count = min_count - current_count
 
+            # Silly sanity check. If count < 0, set it to 0. There is no
+            # concept of launching "negative" instance counts.
+            if count < 0:
+                count = 0
+
         if self._dry:
             self.log.info('Would have launched %s instances of array %s' % (
                           count, array.soul['name']))
             raise gen.Return()
 
-        if count < 0:
+        if count < 1:
```

Comment:
Not sure what you mean by line 573 (since thats above the `__init__`), but I think this line is in the right place. It technically could be re-written to be `if count==0`, but the way its written now also catches if a user calls the method with `count==<some negative number>`. I think its in the right place.